server/channels/store/localcachelayer/role_layer.go

```diff
@@ -60,15 +62,30 @@ func (s LocalCacheRoleStore) GetByNames(names []string) ([]*model.Role, error) {
 	var foundRoles []*model.Role
 	var rolesToQuery []string
 
-	for _, roleName := range names {
+	var toPass []any
+	for i := 0; i < len(names); i++ {
 		var role *model.Role
-		if err := s.rootStore.doStandardReadCache(s.rootStore.roleCache, roleName, &role); err == nil {
-			foundRoles = append(foundRoles, role)
+		toPass = append(toPass, &role)
+	}
+	errs := s.rootStore.doMultiReadCache(s.rootStore.roleCache, names, toPass)
+	for i, err := range errs {
+		if err != nil {
+			if err != cache.ErrKeyNotFound {
+				s.rootStore.logger.Warn("Error in Rolestore.GetByNames: ", mlog.Err(err))
+			}
+			rolesToQuery = append(rolesToQuery, names[i])
 		} else {
-			rolesToQuery = append(rolesToQuery, roleName)
+			gotRole := *(toPass[i].(**model.Role))
+			if gotRole != nil {
```

Comment:
Trying to understand whether this can ever be `nil`. If so, should we handle it as a miss or else?